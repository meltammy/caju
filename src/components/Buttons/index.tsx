import styled from "styled-components";

const Button = styled.button`
  outline: none;
  display: flex;
  align-items: center;
  border: none;
  border-radius: 36px;
  padding: 8px 32px;
  background-color: #64a98c;
  cursor: pointer;
  height: 56px;
  color: #fff;
  box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;
  font-size: 16px;
  font-weight: 600;
`;

export default Button;
