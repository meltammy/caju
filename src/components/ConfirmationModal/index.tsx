import { FC } from "react";
import {
  Description,
  ModalContainer,
  ModalOverlay,
  Title,
  ButtonContainer,
  StyledAsyncButton,
} from "./styles";
import Button from "../Buttons/Button";

interface ConfirmationModalProps {
  title: string;
  description: string;
  confirmationButtonLabel: string;
  isOpen: boolean;
  isLoading?: boolean;
  onClose: () => void;
  onConfirm: () => void;
}

export const confirmationModalTestId = "confirmation-modal";
const ConfirmationModal: FC<ConfirmationModalProps> = ({
  title,
  description,
  confirmationButtonLabel,
  isOpen,
  isLoading,
  onClose,
  onConfirm,
}) => {
  return (
    <ModalOverlay isOpen={isOpen} data-testid={confirmationModalTestId}>
      <ModalContainer>
        <Title>{title}</Title>
        <Description dangerouslySetInnerHTML={{ __html: description }} />
        <ButtonContainer>
          <Button onClick={onClose} disabled={isLoading}>
            Cancelar
          </Button>

          <StyledAsyncButton
            isConfirmation
            onClick={onConfirm}
            isLoading={isLoading}
          >
            {confirmationButtonLabel}
          </StyledAsyncButton>
        </ButtonContainer>
      </ModalContainer>
    </ModalOverlay>
  );
};

export default ConfirmationModal;
