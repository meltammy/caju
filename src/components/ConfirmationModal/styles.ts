import styled from "styled-components";

export const ModalOverlay = styled.div<{ $isOpen: boolean }>`
  display: ${(props) => (props.$isOpen ? "flex" : "none")};
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.5);
  justify-content: center;
  align-items: center;
`;

export const ModalContainer = styled.div`
  background: white;
  border-radius: 8px;
  padding: 20px;
  min-width: 500px;
  max-width: 95vw;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
  z-index: 999;

  @media screen and (max-width: 1200px) {
    min-width: 50vw;
    margin: 1rem;
  }
`;

export const Title = styled.h2`
  margin: 0;
`;

export const Description = styled.p`
  margin: 10px 0 20px;
`;

export const ButtonContainer = styled.div`
  display: flex;
  gap: 1rem;

  @media screen and (max-width: 1200px) {
    flex-direction: column-reverse;
  }

  button {
    width: 100%;
    place-content: center;
  }
`;
