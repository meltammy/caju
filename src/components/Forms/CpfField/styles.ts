import ReactInputMask from "react-text-mask";
import styled from "styled-components";

export const Input = styled(ReactInputMask)`
  width: 100%;
  padding: 0 0.5rem;
  min-height: 2.25rem;
  background-color: ${({ theme }) => theme.colors.white};
  transition: all 0.2s ease-in-out 0s;

  vertical-align: middle;
  border: 1px solid rgba(36, 28, 21, 0.3);
  border-radius: 0.5rem;

  font-size: 1rem;
  line-height: 1.125rem;
  font-weight: normal;

  &:focus {
    border: 1px solid ${({ theme }) => theme.colors.petroleumBlue};
    box-shadow: inset 0 0 0 1px ${({ theme }) => theme.colors.petroleumBlue};
  }
`;
