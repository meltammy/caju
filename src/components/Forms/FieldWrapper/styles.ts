import styled from "styled-components";

export const ErrorMessage = styled.span`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 0.75rem;
`;

export const Container = styled.div`
  width: 100%;
`;

export const Label = styled.label`
  width: 100%;
  font-weight: 600;
`;
