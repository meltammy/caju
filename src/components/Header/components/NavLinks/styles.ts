import styled from "styled-components";

export const NavItems = styled.a`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 1.5rem;
  font-weight: 600;
`;
