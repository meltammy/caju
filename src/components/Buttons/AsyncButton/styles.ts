import styled from "styled-components";
import { Spinner } from "~/components/Spinner";

export const StyledSpinner = styled(Spinner)`
  width: 5px;
  height: 5px;
  border-width: 3px;
`;
