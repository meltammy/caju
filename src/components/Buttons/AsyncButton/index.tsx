import { ComponentProps } from "react";
import Button from "../Button";
import { StyledSpinner } from "./styles";

type Props = {
  isLoading?: boolean;
} & ComponentProps<typeof Button>;

export const asyncButtonTestId = "async-button";
export function AsyncButton({ isLoading, children, disabled, ...rest }: Props) {
  return (
    <Button
      {...rest}
      disabled={isLoading || disabled}
      data-testid={asyncButtonTestId}
    >
      {isLoading ? <StyledSpinner /> : children}
    </Button>
  );
}
