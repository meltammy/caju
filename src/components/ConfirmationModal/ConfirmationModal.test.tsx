import "@testing-library/jest-dom";
import "jest-styled-components";

import { render, fireEvent } from "@testing-library/react";
import ConfirmationModal, { confirmationModalTestId } from ".";
import { ThemeProvider } from "styled-components";
import { theme } from "~/../styles/theme";
import { asyncButtonTestId } from "../Buttons/AsyncButton";

const title = "Confirm Action";
const description = "Are you sure you want to proceed?";
const cancelButtonLabel = "Cancelar";

const renderModal = (
  props: Partial<React.ComponentProps<typeof ConfirmationModal>> = {}
) => {
  const defaultProps = {
    title,
    description,
    confirmationButtonLabel: "Yes",
    isOpen: true,
    onClose: jest.fn(),
    onConfirm: jest.fn(),
    ...props,
  };

  return render(
    <ThemeProvider theme={theme}>
      <ConfirmationModal {...defaultProps} />
    </ThemeProvider>
  );
};

describe("ConfirmationModal component", () => {
  it("should render modal with title and description", () => {
    const { getByText } = renderModal();

    expect(getByText(title)).toBeInTheDocument();
    expect(getByText(description)).toBeInTheDocument();
  });

  it("should calls onClose when cancel button is clicked", () => {
    const onClose = jest.fn();
    const { getByText } = renderModal({ onClose });

    fireEvent.click(getByText(cancelButtonLabel));
    expect(onClose).toHaveBeenCalled();
  });

  it("should calls onConfirm when confirmation button is clicked", () => {
    const onConfirm = jest.fn();
    const { getByText } = renderModal({ onConfirm });

    fireEvent.click(getByText("Yes"));
    expect(onConfirm).toHaveBeenCalled();
  });

  it("should disables buttons when loading", () => {
    const { getByRole, getByTestId } = renderModal({ isLoading: true });

    const cancelButton = getByRole("button", { name: /cancelar/i });
    const confirmButton = getByTestId(asyncButtonTestId);

    expect(cancelButton).toBeDisabled();
    expect(confirmButton).toBeDisabled();
  });

  it("should is not visible when isOpen is false", () => {
    const { getByTestId } = renderModal({ isOpen: false });

    const modal = getByTestId(confirmationModalTestId);
    expect(modal).toHaveStyleRule("display", "none");
  });

  it("should match snapshot when open", () => {
    const { asFragment } = renderModal();
    expect(asFragment()).toMatchSnapshot();
  });
});
