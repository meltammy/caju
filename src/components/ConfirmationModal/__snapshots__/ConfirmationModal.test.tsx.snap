// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal component matches snapshot when open 1`] = `
<DocumentFragment>
  .c5 {
  outline: none;
  display: flex;
  align-items: center;
  border: none;
  border-radius: 36px;
  padding: 8px 32px;
  background-color: #64a98c;
  cursor: pointer;
  height: 56px;
  color: #fff;
  box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;
  font-size: 16px;
  font-weight: 600;
}

.c5:disabled {
  background-color: #ccc;
  opacity: 1;
}

.c0 {
  display: flex;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.5);
  justify-content: center;
  align-items: center;
}

.c1 {
  background: white;
  border-radius: 8px;
  padding: 20px;
  max-width: 400px;
  width: 100%;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
}

.c2 {
  margin: 0;
}

.c3 {
  margin: 10px 0 20px;
}

.c4 {
  display: grid;
  gap: 10px;
  grid-template-columns: 1fr 1fr;
}

.c6 {
  padding: 10px 15px;
  border: none;
  border-radius: 5px;
  cursor: pointer;
  background: #e80537;
  color: white;
  height: 35px;
}

.c6:hover {
  opacity: 0.8;
}

.c6:disabled {
  background: #ccc;
  opacity: 1;
  cursor: progress;
}

<div
    class="c0"
    data-testid="confirmation-modal"
  >
    <div
      class="c1"
    >
      <h2
        class="c2"
      >
        Confirm Action
      </h2>
      <p
        class="c3"
      >
        Are you sure you want to proceed?
      </p>
      <div
        class="c4"
      >
        <button
          class="c5"
        >
          Cancelar
        </button>
        <button
          class="c5 c6"
          data-testid="async-button"
        >
          Yes
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
